src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Windows.targets

```diff
@@ -167,4 +167,72 @@ Copyright (c) .NET Foundation. All rights reserved.
+      <CsWinRTGeneratorValidateWinRTRuntimeAssemblyVersion Condition="'$(ValidateWinRTRuntimeAssemblyVersion)' == ''">true</CsWinRTGeneratorValidateWinRTRuntimeAssemblyVersion>
+      <CsWinRTGeneratorTreatWarningsAsErrors Condition="'$(CsWinRTGeneratorTreatWarningsAsErrors)' == ''">false</CsWinRTGeneratorTreatWarningsAsErrors>
+      <CsWinRTGeneratorMaxDegreesOfParallelism Condition="'$(CsWinRTGeneratorMaxDegreesOfParallelism)' == ''">-1</CsWinRTGeneratorMaxDegreesOfParallelism>
+      <CsWinRTGeneratorStandardOutputImportance Condition="'$(CsWinRTGeneratorStandardOutputImportance)' == ''">High</CsWinRTGeneratorStandardOutputImportance>
+      <CsWinRTGeneratorStandardErrorImportance Condition="'$(CsWinRTGeneratorStandardErrorImportance)' == ''">High</CsWinRTGeneratorStandardErrorImportance>
+      <CsWinRTGeneratorLogStandardErrorAsError Condition="'$(CsWinRTGeneratorLogStandardErrorAsError)' == ''">true</CsWinRTGeneratorLogStandardErrorAsError>
+    </PropertyGroup>
+
+    <!-- Invoke 'cswinrtgen' -->
+    <CsWinRTGenerator
+      ReferenceAssemblyPaths="@(ReferencePath)"
+      OutputAssemblyPath="@(IntermediateAssembly)"
+      InteropAssemblyDirectory="$(CsWinRTGeneratorInteropAssemblyDirectory)"
+      DebugReproDirectory="$(CsWinRTGeneratorDebugReproDirectory)"
+      CsWinRTToolsDirectory="$(CsWinRTToolsDirectory)"
+      UseWindowsUIXamlProjections="$(CsWinRTUseWindowsUIXamlProjections)"
+      ValidateWinRTRuntimeAssemblyVersion="$(CsWinRTGeneratorValidateWinRTRuntimeAssemblyVersion)"
+      TreatWarningsAsErrors="$(CsWinRTGeneratorTreatWarningsAsErrors)"
+      MaxDegreesOfParallelism="$(CsWinRTGeneratorMaxDegreesOfParallelism)"
+      AdditionalArguments="@(CsWinRTGeneratorAdditionalArgument)"
+      StandardOutputImportance="$(CsWinRTGeneratorStandardOutputImportance)"
+      StandardErrorImportance="$(CsWinRTGeneratorStandardErrorImportance)"
+      LogStandardErrorAsError="$(CsWinRTGeneratorLogStandardErrorAsError)">
+      <Output ItemName="CsWinRTGeneratorInteropAssemblyPath" TaskParameter="InteropAssemblyPath" />
+    </CsWinRTGenerator>
+
+    <!-- Set the metadata for the interop .dll item -->
+    <ItemGroup>
+      <CsWinRTGeneratorInteropAssemblyPath Condition="'%(Filename)%(Extension)' == 'WinRT.Interop.dll'">
+        <AssemblyName>WinRT.Interop</AssemblyName>
+        <TargetFrameworkIdentifier>.NETCoreApp</TargetFrameworkIdentifier>
+        <ReferenceOutputAssembly>true</ReferenceOutputAssembly>
+        <IncludeRuntimeDependency>true</IncludeRuntimeDependency>
+        <Private>true</Private>
+        <MSBuildSourceTargetName>InvokeCsWinRTGenerator</MSBuildSourceTargetName>
+        <Platforms>AnyCPU</Platforms>
+        <CopyLocal>true</CopyLocal>
+      </CsWinRTGeneratorInteropAssemblyPath>
+    </ItemGroup>
+
+    <!-- Add the interop .dll to the list of reference paths -->
+    <ItemGroup>
+      <ReferencePath Include="@(CsWinRTGeneratorInteropAssemblyPath)" />
```

Comment:
We need two things here:
- `@(CsWinRTGeneratorInteropAssemblyPath)` needs to be copied to the output directory
- `@(CsWinRTGeneratorInteropAssemblyPath)` also needs to be added to .deps.json

I'm not sure how to do both. I could do it by adding the item to both `ReferencePath`, `UserRuntimeAssembly`, and `IntermediatePath`, but that doesn't seem correct. What is the right way to achieve this?

Reply:
Will need to look into answers for both of these, I'll try to do that tomorrow.

Reply:
Resolving this since we addressed this offline and I've updated the PR. Thank you again! 🙂